pycbc/detector.py

```diff
@@ -322,3 +304,96 @@ def overhead_antenna_pattern(right_ascension, declination, polarization):
+                         scale = 'utc').to_datetime(timezone = None)
+
+        t_gps = np.sum(np.array([t_gps.year - 2034, t_gps.month/12,
+                                 t_gps.day/(4380),
+                                 t_gps.hour/(105120), 
+                                 t_gps.minute/(6307200),
+                                 t_gps.second/(37843200),
+                                 t_gps.microsecond/(37843200*1e-6)]), axis=0)
+
+        n = np.array(range(1, 4))
+        kappa, _lambda_ = 0, 0
+        alpha = 2. * np.pi * t_gps/1 + kappa
+        beta_n = (n - 1) + (2. * np.pi/3) + _lambda_
+        a, L = 1., .1   # units are in AU
+        e = L/(2. * a * np.sqrt(3))
+        _prod_ = a*e*(sin(alpha)*cos(alpha)*sin(beta_n)
+
+        x = a*cos(alpha) + _prod_ - (1 + sin(alpha)**2)*cos(beta_n))
+        y = a*sin(alpha) + _prod_ - (1 + cos(alpha)**2)*sin(beta_n))
+        z = -np.sqrt(3)*a*e*cos(alpha - beta_n)
+        self.location = np.array([x,y,z])
+
+        return self.location
+
+    def plot_orbit(self):
+        from mpl_toolkits.mplot3d import Axes3D
+        import matplotlib.pyplot as plt
+        dec_center = []
+        for i in range(2000):
+          dec_center.append(self.get_pos(self.reference_time + i).mean(axis = 1))
+
+        t = Time(val = self.reference_time, format = 'gps', scale ='utc')
+        sun = coordinates.get_sun(t).transform_to('icrs')
+        earth = coordinates.get_body('earth', t, location = None).transform_to('icrs')
+        sun.representation_type, earth.representation_type ='cartesian', 'cartesian'
+
+        fig = plt.figure()
+        ax = plt.axes(projection = "3d")
+        ax.scatter(np.float32(earth.x), np.float32(earth.y), np.float32(earth.z), marker = 'o')
+        ax.scatter(np.float32(sun.x), np.float32(sun.y), np.float32(sun.z), marker = '+')
+        ax.scatter(dec_center[0], dec_center[1] ,dec_center[2], marker = '*')
+        ax.set_xlabel('X axis (AU)')
+        ax.set_ylabel('Y axis (AU)')
+        ax.set_zlabel('Z axis (AU)')
+
+    def time_delay_from_location(self, other_location, right_ascension,
+                                 declination, t_gps):
+        dec_loc = self.get_pos(t_gps)
+        """signal = coordinates.SkyCoord(ra = right_ascension, dec = declination,
+                                          unit = u.rad, frame = 'gcrs').transform_to('icrs')"""
+
+        dx = np.array([other_location[0] - self.location[0],
+                       other_location[1] - self.location[1],
+                       other_location[2] - self.location[2]])
+
+        """ra_angle = self.gmst_estimate(t_gps) - right_ascension"""
+        cosd = cos(declination)
+        e0 = cosd * cos(right_ascension)
+        e1 = cosd * -sin(right_ascension)
+        e2 = sin(declination)
+        ehat = np.array([e0, e1, e2])
+        return dx.dot(ehat) / constants.c.value
+
+    def time_delay_from_detector(self, other_detector, right_ascension,
```

Comment:
It would be good to not assume this. The best way to do that would be to add a method for both the regular detector class and this one that is somthing like "icrs_position". You can then have this method just call that on both self and other to get their positions and then calculate the distance. Similarly, this method could be identical to the one for the regular detector class, so it works either way (i.e. you can ask the distance from lisa -> ligo or ligo -> lisa.